tests/templates/kuttl/oidc/40_druid.yaml.j2

```diff
@@ -0,0 +1,240 @@
+    listenerClass: external-unstable
+    deepStorage:
+      s3:
+        bucket:
+          inline:
+            bucketName: druid
+            connection:
+              inline:
+                host: minio-druid
+                port: 9000
+                accessStyle: Path
+                credentials:
+                  secretClass: druid-s3-credentials
+{% if test_scenario['values']['s3-use-tls'] == 'true' %}
+                tls:
+                  verification:
+                    server:
+                      caCert:
+                        secretClass: minio-tls-certificates
+{% endif %}
+        baseKey: deep-storage
+    metadataStorageDatabase:
+      dbType: derby
+      connString: jdbc:derby://localhost:1527/var/druid/metadata.db;create=true
+      host: localhost
+      port: 1527
+    zookeeperConfigMapName: druid-znode
+  brokers:
+    roleGroups:
+      default:
+        replicas: 1
+    configOverrides:
+      runtime.properties: &runtime-properties
+        druid.extensions.loadList: >-
+          ["postgresql-metadata-storage",
+          "simple-client-sslcontext",
+          "druid-kafka-indexing-service",
+          "druid-datasketches",
+          "prometheus-emitter",
+          "druid-basic-security",
+          "druid-s3-extensions",
+          "druid-pac4j"]
+
+        # basic authenticator needed for internal authentication among Druid processes
+        # Trying to use the pac4j authenticator in the escalator below leads to 302 errors,
+        # it seems like the Druid processes cannot handle the OIDC authentication flow.
+        druid.auth.authenticator.DruidInternal.type: basic
+        druid.auth.authenticator.DruidInternal.initialInternalClientPassword: '${env:DRUID_INTERNAL_PASSWORD}'
+        druid.auth.authenticator.DruidInternal.skipOnFailure: "true"  # for any non system user, skip to the pac4j authenticator
+
+        # pac4j authenticator
+        druid.auth.authenticator.pac4j.type: pac4j
+        # pac4j common config
+        druid.auth.pac4j.cookiePassphrase: '${env:DRUID_COOKIE_PASSPHRASE}'
+        # OIDC common config
+        druid.auth.pac4j.oidc.clientID: '${env:DRUID_OIDC_CLIENT}'
+        druid.auth.pac4j.oidc.clientSecret: '${env:DRUID_OIDC_CLIENT_SECRET}'
+{% if test_scenario['values']['oidc-use-tls'] == 'true' %}
+        druid.auth.pac4j.oidc.discoveryURI: 'https://keycloak.$NAMESPACE.svc.cluster.local:8443/realms/test/.well-known/openid-configuration'
+{% else %}
+        druid.auth.pac4j.oidc.discoveryURI: 'http://keycloak.$NAMESPACE.svc.cluster.local:8080/realms/test/.well-known/openid-configuration'
+{% endif %}
+        druid.auth.pac4j.oidc.oidcClaim: preferred_username
+
+        druid.auth.authenticatorChain: '["DruidInternal","pac4j"]'
+
+        druid.escalator.type: basic
+        druid.escalator.internalClientUsername: druid_system
+        druid.escalator.internalClientPassword: '${env:DRUID_INTERNAL_PASSWORD}'
+    podOverrides:
+      spec:
+        containers:
```

Comment:
Optional: Can we use yaml anchors here as much as possible (similar to [this](https://github.com/stackabletech/demos/blob/adad03b775be054cbd3f97b5c8d9416063649cc9/stacks/end-to-end-security/trino.yaml#L26))?
Did not check the content to be honest